### PR TITLE
edf-scheduler: make sure the idle task is scheduled first

### PR DESCRIPTION
**Pull request by lyakh**
When slave cores are brought up, their IDC is enabled and it can be
triggered immediately, which then attempts to schedule an EDF task.
This cannot work because at that time the EDF scheduler hasn't
started its idle task yet. Make sure we don't attempt to schedule any
custom EDF tasks before the idle task has started.

### CODE REVIEW

**Comment by paulstelian97** [comment]
What will happen if we get bespoke tasks before starting the main task? Would they be lost or are they in a list to also start after the idle task is registered?

File: src/schedule/edf_schedule.c [hunk 1]
```diff
@@ -124,12 +129,16 @@ static void schedule_edf_task(void *data, struct task *task, uint64_t start,
...
 	/* add task to the list */
 	list_item_append(&task->list, &edf_sch->list);
+	tracev_edf_sch("schedule_edf_task() %p", (long)task);
 
 	task->state = SOF_TASK_STATE_QUEUED;
 
 	irq_local_enable(flags);
 
-	schedule_edf(data);
+	if (cpu_get_id() == PLATFORM_MASTER_CORE_ID ||
```
Comment (lgirdwood) [inline src/schedule/edf_schedule.c]: Can you comment here what we are doing and why.

File: src/schedule/edf_schedule.c [hunk 2]
```diff
@@ -113,6 +116,8 @@ static void schedule_edf_task(void *data, struct task *task, uint64_t start,
 		return;
 	}
 
+	main_started = !list_is_empty(&edf_sch->list);
```
Comment (lgirdwood) [inline src/schedule/edf_schedule.c]: inline comment on what we are checking.

File: src/schedule/edf_schedule.c [hunk 3]
```diff
@@ -124,12 +129,16 @@ static void schedule_edf_task(void *data, struct task *task, uint64_t start,
 
 	/* add task to the list */
 	list_item_append(&task->list, &edf_sch->list);
+	tracev_edf_sch("schedule_edf_task() %p", (long)task);
```
Comment (mmaka1) [inline src/schedule/edf_schedule.c]: Note: if your object has uid (a task has one), you may use also %s and provide task->uid to print the task name and uuid value in the log.

**Comment by tlauda** [comment]
@lyakh Can we close this? I think fix is in #2593.

**Comment by lyakh** [comment]
> @lyakh Can we close this? I think fix is in #2593.

@tlauda I think yes, that patch should be preferred, but can we maybe keep this one open until we finalise reviewing, testing and merging #2593?

**Comment by tlauda** [comment]
> > @lyakh Can we close this? I think fix is in #2593.
> 
> @tlauda I think yes, that patch should be preferred, but can we maybe keep this one open until we finalise reviewing, testing and merging #2593?

Sure.

**Pull request closed without merging**